File: app/src/test/java/com/dominikgaller/alpinebooking/booking/core/domain/tourbooking/ParticipantCountTest.java
package com.dominikgaller.alpinebooking.booking.core.domain.tourbooking;

import com.dominikgaller.alpinebooking.booking.core.domain.tourbooking.exception.InvalidBookingRequestException;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatExceptionOfType;

class ParticipantCountTest {

    @Test
    void value1IsValid() {
        assertThat(new ParticipantCount(1).value()).isEqualTo(1);
    }

    @Test
    void largeValueIsValid() {
        assertThat(new ParticipantCount(100).value()).isEqualTo(100);
    }

    @Test
    void value0Throws() {
        assertThatExceptionOfType(InvalidBookingRequestException.class)
                .isThrownBy(() -> new ParticipantCount(0));
    }

    @Test
    void negativeValueThrows() {
        assertThatExceptionOfType(InvalidBookingRequestException.class)
                .isThrownBy(() -> new ParticipantCount(-5));
    }
}

File: src/test/kotlin/com/example/contractmanagement/booking/core/domain/tourbooking/ParticipantCountTest.kt
package com.example.contractmanagement.booking.core.domain.tourbooking

import com.example.contractmanagement.booking.core.domain.tourbooking.exception.InvalidBookingRequestException
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

class ParticipantCountTest {
    @Test
    fun construction_succeeds_forTheSmallestValidCount() {
        assertThat(ParticipantCount(1).value).isEqualTo(1)
    }

    @Test
    fun construction_throwsInvalidBookingRequestException_whenCountIsZero() {
        assertThatThrownBy { ParticipantCount(0) }
            .isInstanceOf(InvalidBookingRequestException::class.java)
    }

    @Test
    fun construction_throwsInvalidBookingRequestException_whenCountIsNegative() {
        assertThatThrownBy { ParticipantCount(-1) }
            .isInstanceOf(InvalidBookingRequestException::class.java)
    }
}
